importer.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+type Importer struct {
```

Comment:
Is Importer the right object here? Does there even need to be a struct? Presumably importer would handle a variety of other importing things?

An alternative would be passing these objects directly to funcs. Perhaps a signature like `ImportWorkspace(ctx context.Context, tfcli *tfexec.Terraform, tfapi *tfe.Client, ImportWorkspaceRequest)`, where `ImportWorkspaceRequest` has fields for all parameters needed to construct API calls/CLI commands. 

Typically a struct with methods (a receiver) is most necessary when there is shared state across the methods that is not directly exposed. It can make sense to use it just to share dependencies but for a small-is number of dependencies it should be possible to fit into the function signature with a request argument.

Reply:
i see! no we won't have any shared state within the struct. In that light a function will work just fine